fireplace-swing/src/main/java/com/github/bric3/fireplace/flamegraph/FlameGraphPainter.java

```diff
@@ -209,34 +204,34 @@ private void paint(Graphics2D g2, Rectangle visibleRect, boolean minimapMode) {
                                                             || frame.endX < selectedFrame.startX
                                                             || frame.startX > selectedFrame.endX),
                                                     minimapMode),
-                                        frameBorderColor,
+                                        frameGapColor,
                                         minimapMode);
             }
         }
 
         if (!minimapMode) {
-            paintHoveredFrameBorder(g2, visibleRect, frameBoxHeight, rectOnCanvas);
+            paintHoveredFrameBorder(g2d, visibleRect, frameBoxHeight, rectOnCanvas);
         }
 
         if (!minimapMode && paintDetails) {
             // timestamp
             var drawTimeMs = "FrameGraph width " + flameGraphWidth + " Zoom Factor " + zoomFactor + " Coordinate (" + visibleRect.x + ", " + visibleRect.y + ") size (" +
                              visibleRect.width + ", " + visibleRect.height +
                              ") , Draw time: " + (System.currentTimeMillis() - start) + " ms";
-            var nowWidth = g2.getFontMetrics().stringWidth(drawTimeMs);
-            g2.setColor(Color.darkGray);
-            g2.fillRect(visibleRect.x + visibleRect.width - nowWidth - textBorder * 2,
+            var nowWidth = g2d.getFontMetrics().stringWidth(drawTimeMs);
+            g2d.setColor(Color.DARK_GRAY);
+            g2d.fillRect(visibleRect.x + visibleRect.width - nowWidth - textBorder * 2,
                         visibleRect.y + visibleRect.height - frameBoxHeight,
                         nowWidth + textBorder * 2,
                         frameBoxHeight);
 
-            g2.setColor(Color.yellow);
-            g2.drawString(drawTimeMs,
+            g2d.setColor(Color.YELLOW);
+            g2d.drawString(drawTimeMs,
                           visibleRect.x + visibleRect.width - nowWidth - textBorder,
                           visibleRect.y + visibleRect.height - textBorder);
         }
 
-        g2.clip(visibleRect);
+        g2d.dispose();
```

Comment:
The clip is not necessary ?

Reply:
Since `g2` is the instance passed in from outside, you are not suppose to modify it permanently - https://docs.oracle.com/en/java/javase/17/docs/api/java.desktop/javax/swing/JComponent.html#paintComponent(java.awt.Graphics).  Let me try the double-click, I didn't notice a problem with it (and I assumed it just triggered a full repaint which would get a new graphics instance in any case).